### ISSUE TITLE
yolo11 model.track, sometime no masks data

### ISSUE THREAD

**Issue by tisu97** [ultralytics/ultralytics#19459]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hey there YOLO community. I am new to YOLO. Currently, I apply YOLO tracking function for the object tracking. Now I faced a tricky problem.
My sample code is shown below:
```
# Load an official or custom model
model = YOLO("./runs/train/89_20250227/weights/best.pt")  # Load an official Detect model
video_path = r'./videos/20250227/89_1.mp4'

# Perform tracking with the model
results = model.track(video_path, show=False, stream=True)  # Tracking with default tracker
for result in results:
    print(result.masks)
```

After training the model by making use of the pretrained model yolo11n-pt, I gonna make the object tracking work. Sometimes, I can observe the masks and the labels of the objects on the tracking screen and the `result.masks` return relevant data. 

However, sometimes, if I trained another model and applied it for object tracking, I could also see the labels but no masks of objects. Also, no `masks.data` was shown.

I noticed that this situaltion not only happens when `model.track` is running. The same thing happens if `model.predict` is running in my code. I didn't understand why the mask was `None`.

B.R.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tisu97, thank you for your interest in Ultralytics 🚀! We recommend reviewing the [Docs](https://docs.ultralytics.com/) for guidance, where you'll find information about [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. These resources may have answers to similar questions.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it? This will give us the necessary context and allow us to identify any potential issues with the feature.

If this is a ❓ Question, please include additional details like training logs and dataset examples, as well as an explanation of how you trained the model (`yolo11n-pt`), so we can assist you better. Also, ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To investigate further, please make sure to upgrade to the latest version of the `ultralytics` package as issues may already be resolved in newer releases. You can upgrade with the following command:

```bash
pip install -U ultralytics
```

## Debugging and Environment Suggestions

Make sure you are using a compatible and updated environment. YOLO works seamlessly in the following environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud**: See [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Quickstart Guides](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Utilize our [Docker Image](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Our Continuous Integration (CI) system performs daily tests to ensure code reliability. Check the status of the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) system with the badge below:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

CI tests validate YOLO functionality, including all [Modes](https://docs.ultralytics.com/modes/) like Detection and Tracking, across multiple platforms.

If you're encountering inconsistent results with `result.masks`, it could be model-related. We’d recommend rechecking your training settings, ensuring your dataset has sufficient labels for masks, and validating inference settings. Additional screenshots or details would also be helpful!

Feel free to interact with the Ultralytics community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time support
- [Discourse](https://community.ultralytics.com/) for focused discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for broader community sharing and exploration

This is an automated response to expedite your support request, but an Ultralytics engineer will assist you soon. Thank you for your patience and support 🙏!

**Comment by glenn-jocher** [comment]
@tisu97 thank you for your question. For masks to be available during tracking, you must use a segmentation model (YOLO11-seg) and ensure object detections meet the confidence threshold. The intermittent mask availability suggests either:

1. Your custom model was not trained as a segmentation model. Please verify you used a segmentation model architecture (like yolov8n-seg.pt) as your initial weights when training.

2. The confidence scores for some frames/detections fall below the tracker's [`track_high_thresh`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/trackers/bytetrack.yaml#L5) value (default 0.5), causing those detections to be ignored by the tracker.

You can test this immediately by running:
```python
model = YOLO('yolo11n-seg.pt')  # official segmentation model
results = model.track(your_video_path)
print(results[0].masks)  # should now contain masks
```

For more details, see the [Instance Segmentation and Tracking guide](https://docs.ultralytics.com/guides/instance-segmentation-and-tracking/).

**Comment by tisu97** [comment]
> [@tisu97](https://github.com/tisu97) thank you for your question. For masks to be available during tracking, you must use a segmentation model (YOLO11-seg) and ensure object detections meet the confidence threshold. The intermittent mask availability suggests either:
> 
> 1. Your custom model was not trained as a segmentation model. Please verify you used a segmentation model architecture (like yolov8n-seg.pt) as your initial weights when training.
> 2. The confidence scores for some frames/detections fall below the tracker's [`track_high_thresh`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/trackers/bytetrack.yaml#L5) value (default 0.5), causing those detections to be ignored by the tracker.
> 
> You can test this immediately by running:
> 
> model = YOLO('yolo11n-seg.pt')  # official segmentation model
> results = model.track(your_video_path)
> print(results[0].masks)  # should now contain masks
> For more details, see the [Instance Segmentation and Tracking guide](https://docs.ultralytics.com/guides/instance-segmentation-and-tracking/).

@glenn-jocher Thank you so much for your reply! Actually, in the situation where the mask wasn't shown during the tracking, I used `yolov8-p2.yaml` to train the model entirely, because I had to detect, segment and track very small objects in the video stream. 

If I understand you correctly, if I want to track and make the mask shown during the tracking prediction, I need to use the segmentation model architecture like `yolov8n-seg.yaml` instead of `yolov8-p2.yaml` for the entire training?

By the way, the reason I choose `yolov8-p2.yaml` is that it shows better performance in the small object detection. But as you mentioned, this model can't output masks during tracking or segmentation, right?

Look forward to your reply. Thanks in advance!

B.R.

**Comment by tisu97** [comment]
I tried according to the reply from @glenn-jocher Yes, if I want to see the  masks during the prediction/segmentation/tracking, for the training process, I need to use the segmentation architecture such as `yolo11n-seg.pt` or `yolo11-seg.yaml` .

But when it comes to the segmentation/tracking for small objects in the video, is there any good method to try? What I have tried now shows that `yolo11n-seg.pt` or `yolo11-seg.yaml`  cannot perform well on small object segmentation, like 40 pixels large object in a 720p image/video. no small objects were detected.

**Comment by glenn-jocher** [comment]
Thank you for your follow-up! For small object segmentation with YOLO11, we recommend these adjustments:

1. **Increase input resolution** by training/predicting with `imgsz=1280` (or higher) to better capture small objects.  
2. **Modify confidence thresholds**: Lower `conf` and `track_high_thresh` values in your tracker configuration to retain smaller detections.  
3. **Custom train** a segmentation model using `yolo11-seg.yaml` architecture with small-object-optimized parameters like `mask_ratio=4` in your dataset YAML.

For implementation details, see the [Segmentation Model Training guide](https://docs.ultralytics.com/tasks/segment/#train) and [Object Tracking Configuration docs](https://docs.ultralytics.com/modes/track/#configuration).
